### PR TITLE
PET workflows that uses the library in ngen framework in automated testing

### PR DESCRIPTION
**Pull request by stcui007**
This PR add workflows codes to run automatic test for PET library with new commits in the ngen framework with the library. 

## Additions

.github/workflows/ngen_integration.yaml

## Removals

-

## Changes

.github/workflows/build_and_run_tests.yml

## Testing

All automated tests: Run PET in standalone mode, run PET within the ngen framework

## Notes

-

## Todos

-

## Checklist

- [x ] PR has an informative and human-readable title
- [x ] Changes are limited to a single goal (no scope creep)
- [x ] Code can be automatically merged (no conflicts)
- [x ] Code follows project standards (link if applicable)
- [x ] Passes all existing automated tests
- [x ] Any _change_ in functionality is tested
- [X ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [x ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: .github/workflows/ngen_integration.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          mv ${{runner.temp}}/evapotranspiration/* extern/evapotranspiration/evapotranspiration
+
+      - name: Build PET
+        id: submod_build_5
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/evapotranspiration/evapotranspiration"
+          targets: "petbmi"
+
+      - name: Run Ngen Test
```
Comment (madMatchstick) [inline .github/workflows/ngen_integration.yaml:101]: Unclear why this is here, "Run ngen with topmodel with CFE"?  The [`example_bmi_multi_realization_config.json`](https://github.com/NOAA-OWP/ngen/blob/master/data/example_bmi_multi_realization_config.json) couples cfe-noahowp. Recommend removing.

File: .github/workflows/ngen_integration.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        run: |
+          # Run ngen with topmodel with CFE
+          mv ${{ steps.ngen_id1.outputs.build-dir }} ./ngen-build/
+          inputfile='data/example_bmi_multi_realization_config.json'
+          ./ngen-build/ngen ./data/catchment_data.geojson "cat-27" ./data/nexus_data.geojson "nex-26" $inputfile
+
+      - name: Run surfacebmi, cfebmi and petbmi
```
Comment (madMatchstick) [inline .github/workflows/ngen_integration.yaml:108]: This one looks good.

File: .github/workflows/build_and_run_tests.yml
Comment (madMatchstick) [inline .github/workflows/build_and_run_tests.yml:1]: thanks for the version update

**Review by madMatchstick — CHANGES_REQUESTED** [review]

File: .github/workflows/ngen_integration.yaml (hunk 1 above)
Comment (stcui007) [reply .github/workflows/ngen_integration.yaml:101]: This is my oversight. I should have used a realization file in the test that uses the PET. Will change that,

Comment (stcui007) [reply .github/workflows/ngen_integration.yaml:101]: Looks like I did have a second test that uses PET. I can remove the one that does not use PET, but it probably good to have it there.

Comment (madMatchstick) [reply .github/workflows/ngen_integration.yaml:101]: I would just remove it for the time being.  The topmod library isn't even being built in this workflow.  The `Run surfacebmi, cfebmi and petbmi` should suffice.  We can always add more builds later on. -thanks

**Review by madMatchstick — APPROVED** [review]
Thanks for the updates and commits Shengting. Okay to merge.

**Comment by stcui007** [comment]
Thank you for your careful review and good suggestions.

**Pull request merged**